content/projects/gsoc/2020/projects/git-plugin-performance.adoc

```diff
@@ -23,44 +24,94 @@ links:
+The aim of the project is to find and improve potential areas in the git plugin for performance enhancement. The process of doing that was to find differences in performance between the two co-existing implementations for git, `CLI git` and `JGit`.
+It is logical to leverage those differences by creating a feature which would automatically switch git implementations optimally adjusting to a certain scenario or a use-case.
+It is also logical to fix existing performance issues within the git plugin to enhance its overall performance.
+
+The project identified git SCM checkout as the area which showed potential scope for enhancements. Improving any portion of the process meant a direct reduction in the overall execution time of a Jenkins job.
+
+=== Background of the project
+The git plugin is one of the most widely used Jenkins component which provides git source code management functionalities to the Jenkins ecosystem. These functionalities or the git operations (like fetch or checkout) are provided using two mutually exclusive implementations, link:https://github.com/jenkinsci/git-client-plugin/blob/master/src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java[CliGitAPIImpl] and link:https://github.com/jenkinsci/git-client-plugin/blob/master/src/main/java/org/jenkinsci/plugins/gitclient/JGitAPIImpl.java[JGitAPIImpl]. 
+
+By default, only command line git is enabled for a user. To enable JGit and fully appreciate the benefits of the new features, go to this doc: link:https://plugins.jenkins.io/git-client/#enabling-jgit[Enabling JGit in the git plugin]
+
+The task was to compare both of these available implementations for certain repository parameters (like size of git objects, number of branches) in multiple platforms to derive a so called *universal rule* which would help the plugin decide which implementation is optimal in a certain situation.
 
 === Project Deliverables
-* Integration of JMH: micro-benchmark module addition to test module of git plugins
-  ** This includes both utilities to support the creation of a performance benchamrking test environment in git-plugin and benchmark tests written for git operations critical to the plugin's end-to-end performance
-* A new additional behavior in GitSCM checkout in build configuration
-  ** The behavior is supposed to enable performance, which will in turn selectively prioritize CliGitAPIImpl or JGitAPIImpl for the best performance
-* A working fix for link:https://github.com/jenkinsci/git-plugin/pull/845[git double/redundant fetch issue]
-  ** A fix which ensures no repository/user data loss if second fetch is removed with automated tests to confirm the intended behaviour
+
+* Fix the occurrence of a second fetch in the checkout process link:https://github.com/jenkinsci/git-plugin/pull/845[JENKINS-49757]
```

Comment:
Oleg is correct.  The document should link to both the bug report and the pull request.
```suggestion
* Remove a redundant fetch (link:https://issues.jenkins-ci.org/browse/JENKINS-49757[JENKINS-49757]) during initial checkout (link:https://github.com/jenkinsci/git-plugin/pull/845[PR-845])
```